src/server.rs

```diff
@@ -963,6 +965,36 @@ where
     }
 }
 
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct PerLanguageCount {
+    counts: HashMap<String, u64>,
+}
+
+impl PerLanguageCount {
+    fn increment(&mut self, kind: &CompilerKind) {
+        let key = kind.lang_kind().clone();
+        let count = match self.counts.get(&key) {
```

Comment:
FYI, `HashMap` has a great API for exactly this use case:
https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.entry